lib/src/client_io.dart

```diff
@@ -283,7 +283,7 @@ class ClientIO extends ClientBase with ClientMixin {
     while (offset < size) {
       List<int> chunk = [];
       if (file.bytes != null) {
-        final end = min(offset + CHUNK_SIZE - 1, size - 1);
+        final end = min(offset + CHUNK_SIZE, size);
```

Comment:
@lohanidamodar do we really want to make this change?

Reply:
yes